toolchain/parse/handle_paren_expr.cpp

```diff
@@ -38,15 +36,13 @@ static auto HandleParenExprParamFinish(Context& context, bool as_tuple)
   // If this is the first item and a comma was found, switch to tuple handling.
   // Note this could be `(expr,)` so we may not reuse the current state, but
   // it's still necessary to switch the parent.
-  if (!as_tuple) {
-    state.state = State::ParenExprParamFinishAsTuple;
-
-    auto finish_state = context.PopState();
-    CARBON_CHECK(finish_state.state == State::ParenExprFinishAsNormal)
-        << "Unexpected parent state, found: " << finish_state.state;
-    finish_state.state = State::ParenExprFinishAsTuple;
-    context.PushState(finish_state);
-  }
+  state.state = State::TupleLiteralElementFinish;
+
+  auto finish_state = context.PopState();
+  CARBON_CHECK(finish_state.state == State::ParenExprFinish)
+      << "Unexpected parent state, found: " << finish_state.state;
+  finish_state.state = State::TupleLiteralFinish;
+  context.PushState(finish_state);
 
   // On a comma, push another expression handler.
   if (list_token_kind == Context::ListTokenKind::Comma) {
```

Comment:
I found this confusing, given that we always have a comma in this case.
```suggestion
  // If the close paren is not next, push another expression handler.
  if (list_token_kind != Context::ListTokenKind::CommaClose) {
```

Reply:
Done, but with some other comment changes in light of your other comment.